### PR TITLE
Add bloom filters with false-negative tests and FPR measurements

### PR DESCRIPTION
**Pull request by jorisdral**

### CODE REVIEW

File: src/Database/LSMTree/Internal/Run/BloomFilter.hs [hunk 1]
```diff
@@ -22,4 +24,66 @@
...
+    monkeyFPR
+  , monkeyBits
+  , monkeyHashFuncs
+    -- * Bloom
+  , Bloom
+  , Bloom.elem
+    -- ** Hashes
+  , Hash
+  , Hash.Hashable (..)
+  , Hash.cheapHashes
```
Comment (phadej) [inline src/Database/LSMTree/Internal/Run/BloomFilter.hs:38]: I don't think we can use these. As `bloomfilter` package *does not* provide any serialization guarantees, i.e. is a memory only package, it for sure doesn't provide strong guarantees that the `cheapHashes` will have the same implementation indefinitely.

Comment (jorisdral) [reply src/Database/LSMTree/Internal/Run/BloomFilter.hs:38]: Alright, maybe we can "fork" the implementation of `cheapHashes` so we don't have to come up with our own hashing functions. How does that sound?

Comment (jorisdral) [reply src/Database/LSMTree/Internal/Run/BloomFilter.hs:38]: Or, we can add a number of golden tests so we can see when the hash function changes

Comment (phadej) [reply src/Database/LSMTree/Internal/Run/BloomFilter.hs:43]: The file format will still depend on an implementation detail of a package, i.e. cannot be stated independently in a specification (of said file format).

If that is not an issue, it's ok.

I wouldn't use a persistence layer which has such 3rd party implementation dependency myself.

**Review by dcoutts — APPROVED** [review]
@phadej makes a good point about file format stability. We will need to think about that carefully and agree an adequate solution.

**Pull request merged**